src/main/java/com/notably/logic/suggestion/commands/OpenSuggestionCommand.java

```diff
@@ -0,0 +1,117 @@
+public class OpenSuggestionCommand implements SuggestionCommand {
+    public static final String COMMAND_WORD = "open";
+    public static final String RESPONSE_MESSAGE = "Open a note";
+
+    private AbsolutePath path;
+
+    public OpenSuggestionCommand(AbsolutePath path) {
+        Objects.requireNonNull(path);
+        this.path = path;
+    }
+
+    @Override
+    public void execute(Model model) {
+        // Nullity check
+        Objects.requireNonNull(model);
+
+        // Set response text
+        model.setResponseText(RESPONSE_MESSAGE);
+
+        // Set suggestions
+        List<AbsolutePath> possiblePaths = getPossiblePaths(model);
+        List<SuggestionItem> suggestions = getSuggestions(possiblePaths, model);
+
+        setSuggestionsToModel(model, suggestions);
+    }
+
+
+    public void setSuggestionsToModel(Model model, List<SuggestionItem> suggestions) {
+        Objects.requireNonNull(model);
+        Objects.requireNonNull(suggestions);
+
+        model.setSuggestions(suggestions);
+    }
+
+    /**
+     * Gets the list of possible absolute paths, based on the user's input path.
+     * @param model The app's model.
+     * @return List of possible absolute paths, based on the user's input path.
+     */
+    public List<AbsolutePath> getPossiblePaths(Model model) {
+        Objects.requireNonNull(model);
+
+        List<AbsolutePath> possiblePaths = new ArrayList<>();
+
+        BlockTreeItem source = model.getBlockTree().get(path);
+        List<String> components = path.getComponents();
+
+        possiblePaths.add(path);
+
+        List<BlockTreeItem> children = source.getBlockChildren();
+        if (children.size() != 0) {
+            for (BlockTreeItem child : children) {
+                possiblePaths = getChildDfs(child, components, possiblePaths);
+            }
+        }
+
+        return possiblePaths;
+    }
+
+    /**
+     * Traverses the tree block starting from the block with the path that the user inputs.
+     * @param curr Current BlockTreeItem.
+     * @param components The components of the current path.
+     * @param possiblePaths The list of possible paths.
+     * @return The list of possible paths.
+     */
+    private List<AbsolutePath> getChildDfs(BlockTreeItem curr, List<String> components,
+        List<AbsolutePath> possiblePaths) {
+        List<String> newComponents = new ArrayList<>();
+        newComponents.addAll(components);
+        newComponents.add(curr.getTitle().getText());
+
+        List<BlockTreeItem> children = curr.getBlockChildren();
+
+        if (children.size() == 0) { // if last element in THAT current path, add curr path to list of possible paths.
+            AbsolutePath newPath = AbsolutePath.fromComponents(newComponents);
+            possiblePaths.add(newPath);
+        } else {
+            for (BlockTreeItem child : children) {
+                getChildDfs(child, newComponents, possiblePaths);
+            }
+        }
+        return possiblePaths;
+    }
```

Comment:
Nit: If you're already mutating the supplied `possiblePaths`, why would you still return `possiblePaths`? It seems a bit inconsistent to me 😅